File: app/src/main/java/com/shen/somebase/widget/manager_drag_adapter/ITouchHelper.java
package com.shen.somebase.widget.manager_drag_adapter;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

/*
 * created by shen at 2019/10/11 16:15
 */
interface ITouchHelper <T extends RecyclerView.Adapter> {
    boolean isLongDragEnable();
    boolean isSwipeEnable();

    @Nullable
    BaseItemTouchHelpCallBack.OnItemTouchCallbackListener<T> listener();

    @Nullable
    BaseItemTouchHelpCallBack.ItemStatusChange listenerStatus();
}

File: app/src/main/java/com/shen/somebase/widget/manager_drag_adapter_kt/ITouchHelper.kt
package com.shen.somebase.widget.manager_drag_adapter_kt

import androidx.recyclerview.widget.RecyclerView

/**
 * created by shen on 2019/10/13
 * at 16:59
 **/
interface ITouchHelper<T : RecyclerView.Adapter<*>> {
    val isLongDragEnable: Boolean
    val isSwipeEnable: Boolean
    val onSwiped: OnSwiped?
    val onMove: OnMove?
    val onSelectChange: OnSelectChange?
    val onClearView: OnClearView?
}